drivers/sensors/l86xxx_uorb.c

```diff
@@ -0,0 +1,754 @@
+        }
+
+        /* All remaining sentences are not transmitted by the module */
+
+        case MINMEA_SENTENCE_GSV:
+        case MINMEA_SENTENCE_GBS:
+        case MINMEA_SENTENCE_GST:
+        case MINMEA_SENTENCE_ZDA:
+        {
+          break;
+        }
+
+        case MINMEA_INVALID:
+        {
+          break;
+        }
+
+        case MINMEA_UNKNOWN:
+        {
+          break;
+        }
+      }
+
+      nxmutex_unlock(&dev->devlock);
+    }
+
+  return 0;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: l86xxx_register
+ *
+ * Description:
+ *   Register the L86-XXX GNSS device driver.
+ *
+ * Arguments:
+ *    devpath   -  The device path to use for the driver
+ *    uartpath  -  The path to the UART character driver connected to the
+ *                 transceiver
+ *    devno     -  The device number to use for the topic (i.e. /dev/mag0)
+ ****************************************************************************/
+
+int l86xxx_register(FAR const char *devpath,
+              FAR const char *uartpath, int devno)
+{
+  FAR l86xxx_dev_s *priv = NULL;
+  int err;
+
+  DEBUGASSERT(devpath != NULL);
+  DEBUGASSERT(uartpath != NULL);
+
+  /* Initialize device structure */
+
+  priv = kmm_zalloc(sizeof(l86xxx_dev_s));
+  if (priv == NULL)
+    {
+      snerr("Failed to allocate instance of L86-XXX driver.\n");
+      return -ENOMEM;
+    }
+
+  memset(priv, 0, sizeof(l86xxx_dev_s));
+
+  /* Initialize mutex */
+
+  err = nxmutex_init(&priv->devlock);
+  if (err < 0)
+    {
+      snerr("Failed to initialize mutex for L86-XXX device: %d\n", err);
+      goto free_mem;
+    }
+
+  /* Initialize semaphore */
+
+  err = nxsem_init(&priv->run, 0, 0);
+  if (err < 0)
+    {
+      snerr("Failed to register nau7802 driver: %d\n", err);
+      goto destroy_mutex;
+    }
+
+  /* Open UART interface for use */
+
+  err = file_open(&priv->uart, uartpath, O_RDWR | O_CLOEXEC);
+  if (err < 0)
+    {
+      wlerr("Failed to open UART interface %s for L86-XXX driver: %d\n",
+            uartpath, err);
+      goto destroy_sem;
+    }
+
+  /* Setup sensor with configured settings */
+
+  read_line(priv); /* Wait until module is powered on */
+
+  #ifdef CONFIG_SERIAL_TERMIOS
+  err = send_command(priv, SET_NMEA_BAUDRATE, L86_XXX_BAUD_RATE);
+  if (err != 3)
+    {
+      snwarn("Couldn't set baud rate of device: %d\n", err);
+    }
+
+  err = send_command(priv, SET_POS_FIX, L86_XXX_FIX_INT);
+  if (err != 3)
+    {
+      snwarn("Couldn't set position fix interval, %d\n", err);
+    }
+
+  #endif
+
+  /* Register UORB Sensor */
+
+  priv->lower.ops = &g_sensor_ops;
+  priv->lower.type = SENSOR_TYPE_GNSS;
+
+  err = sensor_register(&priv->lower, devno);
```

Comment:
gps is a too common to have the dedicated support in uorb, so it's better to use gnss_lowerhalf_s:
https://github.com/apache/nuttx/blob/master/include/nuttx/sensors/gnss.h
https://github.com/apache/nuttx/blob/master/drivers/sensors/gnss_uorb.c
which could simplify gps driver development.

Reply:
This should really be documented

Reply:
I wasn't aware of this support for GNSS modules prior to beginning work on this driver. Despite that, I believe that the driver is good enough that I'm fine keeping it as it is.